modules/search-api/src/main/scala/io/renku/search/api/tapir/Params.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 Swiss Data Science Center (SDSC)
+ * A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+ * Eidgenössische Technische Hochschule Zürich (ETHZ).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.renku.search.api.tapir
+
+import sttp.tapir.{query as queryParam, *}
+import io.renku.search.api.data.*
+import io.renku.search.query.Query
+import io.renku.search.http.borer.TapirBorerJson
+
+object Params extends TapirCodecs with TapirBorerJson {
+
+  val query: EndpointInput[Query] =
+    queryParam[Query]("q").description("User defined search query").default(Query.empty)
+
+  val pageDef: EndpointInput[PageDef] = {
+    val page =
+      queryParam[Int]("page")
+        .validate(Validator.min(1))
+        .description("The page to retrieve, starting at 1")
+        .default(1)
+
+    val perPage =
+      queryParam[Int]("per_page")
+        .description("How many items to return for one page")
+        .validate(Validator.min(1))
```

Comment:
Just so we won't forget, maybe we could add max validator as we are working on this now.